src/seismometer/data/loader/prediction.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import warnings
+
+import numpy as np
+import pandas as pd
+import pyarrow.parquet as pq
+
+import seismometer.data.pandas_helpers as pdh
+from seismometer.configuration import ConfigProvider
+
+logger = logging.getLogger("seismometer")
+
+
+def parquet_loader(config: ConfigProvider) -> pd.DataFrame:
+    """
+    Load the predictions frame from a parquet file based on config.prediction_path.
+
+    Will restrict the loaded columns to those specified in config.features, if present.
+
+    Parameters
+    ----------
+    config : ConfigProvider
+        The loaded configuration object.
+
+    Returns
+    -------
+    pd.DataFrame
+        The predictions dataframe.
+    """
+    if config.features:  # no features == all features
+        desired_columns = set(config.prediction_columns)
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=FutureWarning)
+            present_columns = set(pq.ParquetDataset(config.prediction_path, use_legacy_dataset=False).schema.names)
+
+        if config.target in present_columns:
+            desired_columns.add(config.target)
+
+        actual_columns = desired_columns & present_columns
+        _log_column_mismatch(actual_columns, desired_columns, present_columns)
+
+        dataframe = pd.read_parquet(config.prediction_path, columns=actual_columns)
+    else:
+        dataframe = pd.read_parquet(config.prediction_path)
```

Comment:
Since the smaller section is the else, can we make that go first?

It also matches your comment better (# no features == all features)

Reply:
I like it, makes me feel a tiny bit better about having an else-statement too